packages/pwa-kit-extension-sdk/src/configs/babel/plugin-application-extensions.ts

```diff
@@ -0,0 +1,106 @@
+                const {installed} = state.opts
+
+                // This is analogus to the work we did in webpack to have aliases for the extensions.
+                const aliases: any = buildAliases(installed)
+                const source = path.node.source.value
+
+                // Check for alias
+                for (const alias in aliases) {
+                    if (source.startsWith(alias)) {
+                        const newPath = source.replace(alias, aliases[alias])
+                        path.node.source = t.stringLiteral(newPath)
+                    }
+                }
+            },
+            Program(path: any, state: any) {
+                const filePath = state.file.opts.filename
+                const {installed, configured, target} = state.opts
+
+                // Add a marker to the state to prevent reprocessing
+                if (processedFiles.has(filePath)) {
+                    // If the file has been processed, skip further transformations
+                    return
+                }
+
+                // Check if the file matches one of the files we want to replace
+                if (filePath.endsWith(extensionsPlaceholderFile)) {
+                    const newContent = renderTemplate({
+                        installed,
+                        configured,
+                        target
+                    })
+
+                    let parsedAst
+
+                    try {
+                        // Parse the new content as a full program
+                        parsedAst = parser.parse(newContent, {
+                            sourceType: 'module', // Ensure it supports import/export
+                            plugins: ['jsx', 'typescript'] // Add additional plugins if needed
+                        }).program
+                    } catch (error: any) {
+                        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                        throw new Error(`Failed to parse content for ${filePath}: ${error.message}`)
+                    }
+
+                    // Mark the file as processed by adding its path to the Set
+                    processedFiles.add(filePath)
+
+                    // Replace the entire file content with the parsed AST
+                    path.replaceWith(parsedAst)
+
+                    // Manually transpile the code to ES5 using Babel
+                    const transpiledCode = babel.transformFromAstSync(parsedAst, newContent, {
+                        filename: filePath,
+                        presets: [
+                            [
+                                require('@babel/preset-env'),
+                                {
+                                    targets: {
+                                        node: 'current'
+                                    }
+                                }
+                            ]
+                        ] // Transpile to ES5
+                    }).code
```

Comment:
This is unnecessary since babel runs plugin first and **then** transpile the code for the target environment via the presets. The babel config already uses the presets, this duplicates the job.